timon/probes.py

```diff
@@ -444,19 +446,18 @@ def parse_result(self, jsonstr):
         logger.debug("resp %r", resp)
 
         exit_code = rslt['exit_code']
-        if exit_code == flags.FLAG_UNKNOWN:
-            return
         self.msg = resp.get('msg') or rslt.get('reason') or ''
-        if rslt.get("status") != 200:
-            self.status = " UNKNOWN"
+        if exit_code != flags.FLAG_OK:
+            self.status = flags.INV_FLAG_MAP.get(exit_code, "ERROR")
+            return
         elif self.match_rule(rslt, self.ok_rule):
             self.status = "OK"
         elif self.match_rule(rslt, self.warning_rule):
             self.status = "WARNING"
         elif self.match_rule(rslt, self.error_rule):
             self.status = "ERROR"
         else:
-            self.status = "UNKNOWN"
+            self.status = "ERROR"
```

Comment:
if explicit error rule is not matched I'm not sure we should return ERROR

Reply:
removed